src/selectors/entities/channels.js

```diff
@@ -525,30 +550,34 @@ export const getSortedUnreadChannelIds = createIdsSelector(
             }
 
             return c;
-        }).filter((c) => c).sort((a, b) => {
-            const aMember = myMembers[a.id];
-            const bMember = myMembers[b.id];
-            const aIsMention = a.type === General.DM_CHANNEL || (aMember && aMember.mention_count > 0);
-            let bIsMention = b.type === General.DM_CHANNEL || (bMember && bMember.mention_count > 0);
-
-            if (lastUnreadChannel && b.id === lastUnreadChannel.id && lastUnreadChannel.hadMentions) {
-                bIsMention = true;
-            }
-
-            if (aIsMention === bIsMention && isChannelMuted(bMember) === isChannelMuted(aMember)) {
-                return sortChannelsByDisplayName(locale, a, b);
-            } else if (aIsMention || (isChannelMuted(bMember) && !isChannelMuted(aMember))) {
-                return -1;
-            }
-
-            return 1;
         });
 
-        return allUnreadChannels.map((c) => c.id);
+        return allUnreadChannels;
     }
 );
 
-export const getSortedFavoriteChannelWithUnreadsIds = createIdsSelector(
+export const getMapAndSortedUnreadChannelIds = createIdsSelector(
+    getUnreadChannels,
+    getCurrentUser,
+    getMyChannelMemberships,
+    getLastPostPerChannel,
+    (state, lastUnreadChannel, sorting = 'alpha') => sorting,
+    mapAndSortChannelIds,
+);
+
+export const getSortedUnreadChannelIds = createIdsSelector(
+    getUnreadChannelIds,
+    getFavoritesPreferences,
+    (state, lastUnreadChannel, unreadsAtTop, favoritesAtTop, sorting = 'alpha') => {
+        return getMapAndSortedUnreadChannelIds(state, lastUnreadChannel, sorting);
+    },
+    (unreadChannelIds, favoritePreferences, mappedAndSortedUnreadChannelIds) => {
+        return filterChannels(unreadChannelIds, favoritePreferences, mappedAndSortedUnreadChannelIds, false, false);
```

Comment:
Since we're always passing false into these last two, can we skip that call entirely?